packages/components/src/ui/radix-dropdown/index.tsx

```diff
@@ -0,0 +1,177 @@
+/**
+ * External dependencies
+ */
+import * as DropdownMenuPrimitive from '@radix-ui/react-dropdown-menu';
+
+/**
+ * WordPress dependencies
+ */
+import { forwardRef } from '@wordpress/element';
+import { SVG, Circle } from '@wordpress/primitives';
+import { check, lineSolid } from '@wordpress/icons';
+
+/**
+ * Internal dependencies
+ */
+import Icon from '../../icon';
+import * as DropdownMenuStyled from './styles';
+import type {
+	DropdownMenuProps,
+	DropdownSubMenuProps,
+	DropdownItemProps,
+} from './types';
+
+// Observations / Questions:
+// - is it enough on the larger components to have only one forwarded ref?
+//   If we have only one, should it be to the "root", the "trigger", or the "content"?
+// - Should we be consistent in using the same value for the `asChild` prop on:
+//   - trigger
+//   - content
+//   - sub trigger
+//   - sub content
+// - Should we allow customizing the `asChild` prop on trigger and content?
+// - Props & customisability:
+//   - Which props should we expose?
+//   - Should props be "namespaced" for each subcomponent?
+//   - We should probably explicitly `Pick<>` every prop that we want to expose
```

Comment:
- Keep initial props as low as possible
- Explicitly reference each prop type through square bracket notation, adding our own JSDocs
- We will need some glue code if we want to expose "slot name"-style portal containers

We can reconsider once we have more feedback from other folks on using this component

Reply:
@ciampo to take action